arcgis-ios-sdk-samples/Edit data/Edit features with feature-linked annotation/EditFeaturesWithFeatureLinkedAnnotationViewController.swift

```diff
@@ -153,7 +153,7 @@ class EditFeaturesWithFeatureLinkedAnnotationViewController: UIViewController {
         }
         alert.addAction(doneAction)
         // Add an observer to ensure the user does not input an empty string.
-        textFieldObserver = NotificationCenter.default.addObserver(forName: UITextField.textDidChangeNotification, object: nil, queue: .main, using: {_ in
+        textFieldObserver = NotificationCenter.default.addObserver(forName: UITextField.textDidChangeNotification, object: nil, queue: .main, using: { [unowned alert] _ in
             // Enable the done button if both textfields are not empty.
             doneAction.isEnabled = (alert.textFields?.allSatisfy { $0.text?.isEmpty == false })!
```

Comment:
I'm a little bit unsure. 
Do we also need to capture the `doneAction` here? When it comes to observers I'm always lost on their life cycles... 🤯

Reply:
I think since those actions are dependent on the alert, I don't think there's a need to capture them. Could you confirm @philium ?

Reply:
> Do we also need to capture the `doneAction` here?

Good thought. `doneAction` _is_ being captured here. The question is, how should it be captured? To answer that, consider 1) who has a strong reference to it, and 2) who has a strong reference to those who have a strong reference to it. Here, the observer is retaining the done action. But who is retaining the observer? The Done and Cancel actions? Yes and no. They are holding a reference to the `textFieldObserver` variable, whose value is the observer. Since the variable holds a strong reference to the observer, the actions are as well. There are two different ways to address that:

1) Make the `doneAction` here `unknown`, or
2) Set `textFieldObserver` to `nil` after removing the observer.

Both options break the retain cycle, but they do so in different ways. The first option prevents the retain cycle from being created in the first place. The second option causes the observer to go away when either button is tapped, breaking the retain cycle and allowing the Done button to go away as well.

Does that make sense?

Reply:
I made `doneAction` unowned. Please review once again!